### PR TITLE
Creating get bb server changes function

### PR DESCRIPTION
**Pull request by scott-pickthorn**
Created `Get-BBServerChanges` function for getting changes between two commits or refs.

### CODE REVIEW

File: BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1 [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    )
+    Set-StrictMode -Version 'Latest'
+    $lastPage = $false
+    $nextPageStart = 0
+    $changes = $null
+    $changesList = @()
+
+    while ( -not $lastPage )
+    {
+        $resourcePath = ('projects/{0}/repos/{1}/compare/changes?from={2}&to={3}' -f $ProjectKey, $RepoName, $From, $To)
```
Comment (splatteredbits) [inline BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1]: Make sure to URL encode the `$From` and `$To` parameters, e.g. `[Web.HttpUtility]::UrlEncode($To)`.

File: BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1 [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        $changes = Invoke-BBServerRestMethod -Connection $Connection -Method Get -ApiName 'api' -ResourcePath ('{0}&limit={1}&start={2}' -f $resourcePath, [int16]::MaxValue, $nextPageStart)
+        if( $changes)
+        {
+            $lastPage = $changes.isLastPage
+            $nextPageStart = $changes.nextPageStart
+            $changesList += $changes.values
```
Comment (splatteredbits) [inline BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1]: Don't add the changes to a list. Just return `$changes.values` to the pipeline, e.g.

    $changes.values

File: BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1 [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        }
+        else
+        {
+            return
+        }
+    }
+    return $changesList
```
Comment (splatteredbits) [inline BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1]: Remove.

File: BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1 [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        [string]
+        # name of the target commit or Ref.
+        $To
+
+    )
+    Set-StrictMode -Version 'Latest'
+    $lastPage = $false
+    $nextPageStart = 0
+    $changes = $null
+    $changesList = @()
```
Comment (splatteredbits) [inline BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1]: Remove (see comment below).

File: BitbucketServerAutomation/BitbucketServerAutomation.psd1 [hunk 5]
```diff
@@ -137,11 +138,7 @@ The Bitbucket Server Automation module is used to interact with Bitbucket Server
 
             # ReleaseNotes of this module
             ReleaseNotes = @'
-* Created `Get-BBServerPullRequest` function for getting pull requests.
-* Created `Merge-BBServerPullRequest` function for merging a pull request.
-* Created `New-BBServerPullRequest` function for creating a pull request.
-* Renamed `Get-BBServerFile` function's `FilePath` and `FileName` parameters to `Path` and `Filter`, respectively, to make it clearer what each parameter does.
-* `Get-BBServerFile` now filters on the whole path, not just the file name. 
+* Created `Get-BBServerChanges` function for getting changes between two commits or refs.
```
Comment (splatteredbits) [inline BitbucketServerAutomation/BitbucketServerAutomation.psd1]: Remove the ` or refs` clause.

**Review by splatteredbits — CHANGES_REQUESTED** [review]

File: BitbucketServerAutomation/BitbucketServerAutomation.psd1 (hunk 5 above)
Comment (scott-pickthorn) [reply BitbucketServerAutomation/BitbucketServerAutomation.psd1]: Done

File: BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1 (hunk 3 above)
Comment (scott-pickthorn) [reply BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1]: Done

File: BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1 (hunk 2 above)
Comment (scott-pickthorn) [reply BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1]: Done

File: BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1 (hunk 1 above)
Comment (scott-pickthorn) [reply BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1]: Done

File: BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1 (hunk 4 above)
Comment (scott-pickthorn) [reply BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1]: Done

File: BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1 [hunk 6]
```diff
@@ -56,22 +56,18 @@ function Get-BBServerChanges
     $lastPage = $false
     $nextPageStart = 0
     $changes = $null
-    $changesList = @()
 
     while ( -not $lastPage )
     {
-        $resourcePath = ('projects/{0}/repos/{1}/compare/changes?from={2}&to={3}' -f $ProjectKey, $RepoName, $From, $To)
+        $resourcePath = ('projects/{0}/repos/{1}/compare/changes?from={2}&to={3}' -f $ProjectKey, $RepoName, [Web.HttpUtility]::UrlEncode($From), [Web.HttpUtility]::UrlEncode($To))
```
Comment (splatteredbits) [inline BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1]: Will you add a test for this? Create from and to branches that have a character that needs to be encoded, e.g. `feature/one`, `release/major`, etc.

**Review by splatteredbits — CHANGES_REQUESTED** [review]

File: BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1 (hunk 6 above)
Comment (scott-pickthorn) [reply BitbucketServerAutomation/Functions/Get-BBServerChanges.ps1]: Done

**Review by splatteredbits — APPROVED** [review]

**Pull request merged**